src/main/java/org/broadinstitute/hellbender/utils/collections/NestedIntegerArray.java

```diff
@@ -82,6 +82,64 @@ public T get(final int... keys) {
         return (T)myData[keys[numNestedDimensions]];
     }
 
+    @SuppressWarnings("unchecked")
+    private T leaf(final int key, final Object[] array){
+        return array == null ? null : (T) array[key];
```

Comment:
Yeah, the delicate beauty and elegance of the `incrementDatumOrPutIfNecessary2keys()`, `incrementDatumOrPutIfNecessary3keys()`, and `incrementDatumOrPutIfNecessary4keys()` methods might be dirtied slightly by that extra catch statement :)

Up to you, though!